course_discovery/apps/course_metadata/data_loaders/api.py

```diff
@@ -952,6 +965,8 @@ def _make_request(self, page):
             settings.WORDPRESS_APP_AUTH_PASSWORD
         )
         response = requests.get(self.api_url, auth=auth, params=params)
+        self.api_client.close()
```

Comment:
since, we're not using the client here this will not help. 
we can use a context manager over here:
```
with requests.get(self.api_url, auth=auth, params=params) as response:
    response.raise_for_status()
    return response.json()